sdk/core/azure-core/src/main/java/com/azure/core/util/polling/LocationPollingStrategy.java

```diff
@@ -123,28 +125,30 @@ public Mono<PollResponse<T>> onInitialResponse(Response<?> response, PollingCont
     @Override
     public Mono<PollResponse<T>> poll(PollingContext<T> pollingContext, TypeReference<T> pollResponseType) {
         HttpRequest request = new HttpRequest(HttpMethod.GET, pollingContext.getData(PollingConstants.LOCATION));
-        return httpPipeline.send(request, this.context).flatMap(response -> {
-            HttpHeader locationHeader = response.getHeaders().get(PollingConstants.LOCATION);
-            if (locationHeader != null) {
-                pollingContext.setData(PollingConstants.LOCATION, locationHeader.getValue());
-            }
-
-            LongRunningOperationStatus status;
-            if (response.getStatusCode() == 202) {
-                status = LongRunningOperationStatus.IN_PROGRESS;
-            } else if (response.getStatusCode() >= 200 && response.getStatusCode() <= 204) {
-                status = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
-            } else {
-                status = LongRunningOperationStatus.FAILED;
-            }
-
-            return response.getBodyAsByteArray().map(BinaryData::fromBytes).flatMap(binaryData -> {
-                pollingContext.setData(PollingConstants.POLL_RESPONSE_BODY, binaryData.toString());
-                Duration retryAfter = ImplUtils.getRetryAfterFromHeaders(response.getHeaders(), OffsetDateTime::now);
-                return PollingUtils.deserializeResponse(binaryData, serializer, pollResponseType)
-                    .map(value -> new PollResponse<>(status, value, retryAfter));
+        return FluxUtil.withContext(context1 -> httpPipeline.send(request,
+                CoreUtils.mergeContexts(context1, this.context)))
```

Comment:
Basically all 4 changes is this:

Merge azure-core Context and reactor Context, when sending the request.